### PR TITLE
feat(cli): add `--release` for `android dev`

### LINKED ISSUE #6594: [bug] `window.__TAURI_IPC__ is not a function` when not using a dev server
### Describe the bug

I would like to debug on Android without dev server to see how a release build works like. I can see that it gets the right js script from the asset manager, but `__TAURI_IPC__` is not injected.

By the way, the release build of the same app exits abormally just after starts.

### Reproduction

```
npm run build
cargo tauri android dev --feature custom-protocol
```

### Expected behavior

It should work.

### Platform and versions

```text
Environment
  › OS: Windows 10.0.22621 X64
  › Webview2: 111.0.1661.54
  › MSVC: 
      - Visual Studio Community 2022
  › Node.js: 18.15.0
  › npm: 9.5.0
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.2
  › rustc: 1.70.0-nightly
  › cargo: 1.70.0-nightly
  › Rust toolchain: nightly-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 2.0.0-alpha.4
  › @tauri-apps/api [NPM]: 2.0.0-alpha.1
  › tauri [RUST]: 2.0.0-alpha.4 (no manifest),
  › tauri-build [RUST]: 2.0.0-alpha.2 (no manifest),
  › tao [RUST]: 0.18.3 (no manifest),
  › wry [RUST]: 0.27.1 (no manifest),

App
  › distDir: ../dist
  › devPath: http://localhost:5173/
  › framework: Vue.js
  › bundler: Vite

App directory structure
  ├─ dist
  ├─ node_modules
  ├─ public
  ├─ src
  └─ src-tauri
```

### Stack trace

```text
03-30 22:55:20.977  9079  9079 W Tauri/Console: File: http://127.0.0.1:37361/assets/index.217dae30.js - Line 1 - Msg: Could not find "window.__TAURI_METADATA__". The "appWindow" value will reference the "main" window label.
03-30 22:55:20.977  9079  9079 W Tauri/Console: Note that this is not an issue if running this frontend on a browser instead of a Tauri window.
03-30 22:55:21.046  9079  9079 E Tauri/Console: File: http://127.0.0.1:37361/assets/index.217dae30.js - Line 1 - Msg: TypeError: window.__TAURI_IPC__ is not a function
03-30 22:55:21.053  9079  9079 E Tauri/Console: File: http://127.0.0.1:37361/assets/index.217dae30.js - Line 1 - Msg: TypeError: window.__TAURI_IPC__ is not a function
```

### Additional context

I redirect the dist url to a custom axum server, and the server searches the files from the asset manager. I'm sure that it resolves the right asset files.

I also tries the same app and build features on Windows, and it works well.

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: core/tauri-build/src/mobile.rs [hunk 1]
```diff
@@ -152,8 +137,6 @@ pub fn inject_android_project(
 }
 
 fn copy_folder(source: &Path, target: &Path, ignore_paths: &[&str]) -> Result<()> {
-  let _ = remove_dir_all(target);
```
Comment (lucasfernog) [inline core/tauri-build/src/mobile.rs]: Not removing the entire folder is problematic, it might leave some files that we don't want anymore. Also this is unrelated to the PR.

Comment (amrbashir) [reply core/tauri-build/src/mobile.rs]: `android dev --release` didn't work unless I remove this, because `build` folder was being used while tauri's build script was running and it crashed every time

Comment (lucasfernog) [reply core/tauri-build/src/mobile.rs]: doesn't make much sense, it should also happen without the --release flag. I just tried it and it works fine. Might be some other issue.

Comment (amrbashir) [reply core/tauri-build/src/mobile.rs]: maybe something else on my system locked the directory, lets move on without it then

**Review by lucasfernog — APPROVED** [review]
